src/compiler/factory.ts

```diff
@@ -2587,18 +2587,28 @@ namespace ts {
         return node;
     }
 
-    export function createUnparsedSourceFile(text: string, map?: string): UnparsedSource {
+    export function createUnparsedSourceFile(text: string, mapPath?: string, map?: string): UnparsedSource {

```

Comment:
To avoid possible issues with existing calls, and since these aren't actually optional, we should have the types be `string | undefined` just to make sure we didn't miss any calls.